src/HotChocolate/Core/src/Types/Types/Contracts/IDirective.cs

```diff
@@ -17,6 +18,8 @@ public interface IDirective
 
         T ToObject<T>();
 
+        T ToObject<T>(Type type);
```

Comment:
@michaelstaib , would this be a problem?

I have a scenario where I'd like to be able to loop the directives of a field, finding those that implement an interface, but then actually get the directive class, but `.ToObject<T>` doesn't cut it as T in my case would be an interface and the Activator code fails. Have a look at how it's used in `SchemaExtensions::GetDirectivesWhereRuntimeTypeImplements`

Reply:
Usage:
`directive.ToObject<IXyzDirective>(directive.Type.RuntimeType)`